extensibility/ember-storage/src/extensions.ts

```diff
@@ -0,0 +1,33 @@
+import { type Scope } from 'ember-polaris-service';
+import type Storage from './index.ts';
+
+export interface Registry {}
+
+type Extended<Ext> = typeof Storage & {
+  new (scope: Scope): Storage & Ext;
+};
+
+const Extensions = new WeakMap<typeof Storage, Set<keyof Registry>>();
```

Comment:
On second thought, using a WeakMap here is not quite right; when a class implements/enables a feature, its subclass should also be considered an implementor, so something on the prototype inheritance chain works better. That's a small detail for production consideration, doesn't really affect the overall point this demo is trying to make